src/validations/product/product.validation.js

```diff
@@ -0,0 +1,31 @@
+import Joi from 'joi';
+import {Categories} from '../../database/models/index';
+
+const productValidationSchema=Joi.object({
+    name: Joi.string().min(3).required(),
+    price: Joi.number().integer(),
+    category:Joi.string().required(),
+    expiryDate:Joi.string().required(),
+    bonus: Joi.number().integer(),
+});
+
+const productValidation = async (req, res, next) => {
+    const value = productValidationSchema.validate(req.body, { abortEarly: false });
+    if (value.error) {
+        
+        return res.status(400).json({
+            message:'Invalid product details provided',error:value.error.message
+        });
+    } else {
+        next();
+    }
+};
+export const validateCategory=async(req,res,next)=>{
+    const category=req.body.category.toLowerCase();
+    const cat=await Categories.findOne({where:{name:category}});
+    if(!cat)
+    return res.status(404).json({message:'Invalid Product Category'});
+    req.body.categoryId=cat.id;
+    next();
+};
```

Comment:
This is middleware, not validation. We need to have validation for category separated with product